REFRAME.md

```diff
@@ -0,0 +1,290 @@
+    }
+```
+
+#### Identify
+
+A message for discovering which messages a server supports. May be used by applications to optimize which servers get sent which types of requests.
+
+```ipldsch
+    type IdentifyRequest struct {}
+
+    type IdentifyResponse struct {
+        Methods [String]
+    }
+```
+
+`IdentifyResponse` should return the set of supported methods aside from the "Identify" method.
+
+##### DAG-JSON Examples
+
+Request:
+```
+{"IdentifyRequest" : {}}
+```
+
+Response:
+```
+{"IdentifyResponse" : {
+    "Methods" : ["FindProviders", "GetIPNS"]
+}}
+```
+
+#### FindProviders
+
+A message for finding nodes that have an interest in a given key. Some common examples include finding which peers have advertised that they have a given CID, or which peers are interested in a given pubsub topic.
+
+```ipldsch
+    type FindProvidersRequest struct 
+        Key &Any
+    }
+```
+
+Note: While the Key is a CID it is highly recommended that server implementations treat these FindProvidersRequests as if they were for the multihash. This allows consumers that perform lookups by CID but with a different IPLD codec than one that is known to the server to still get back valid responses.
+
+```ipldsch
+    type FindProvidersResponse struct {
+        Providers [Provider]
+    }
+
+    type Provider struct {
+     Node Node
+     Proto optional [TransferProtocol]
+     }
+
+    # Node describes a node (identity, network address, authentication, whatever else applies)
+    # We expect different types of nodes, e.g. peer, miner, public IP, etc.
+    type Node union {
+     | Peer "peer"
+    } representation keyed
+
+    type Peer struct {
+        ID Bytes # The binary representation of a libp2p peerID https://github.com/libp2p/specs/blob/f433ad595224cf33d916c166d1738f11aadfa9f7/peer-ids/peer-ids.md
+        Multiaddresses optional [Bytes] # Each element in the list is the binary representation of a complete multiaddr without a peerID suffix
+                                        # https://github.com/multiformats/multiaddr/blob/d02c681171629912dd977f4f5d4b3b72dcac507f/README.md
+                                        # https://github.com/libp2p/specs/blob/98e0ca0e54fd83f7b8c86d7e9677e1e430da5810/addressing/README.md#multiaddr-in-libp2p
+    }
+    
+    # An alternative mechanism might be to have a union keyed by the protocol identifier that has more detailed
+    # information inside the bytes. 
+
+    # Note: This is not quite valid IPLD Schema because the "structured" represenation does not exist yet, nor does the SingletonInt mechanism
+    # However, we can use this to describe that we have a union keyed by numbers rather than strings.
+    type TransferProtocol union {
+        | Bitswap
+        | GraphSync-FILv1
+    } representation structured 
+    
+    type Bitswap struct {
+        ID SingletonInt(0x0900) # https://github.com/multiformats/multicodec/blob/f5dd49f35b26b447aa380e9a491f195fd49d912c/table.csv#L133
+    }

```

Comment:
that seems reasonable / this would change once we decide on a deprecation policy.
From what I can see of current implementations, they support downgrading back to all previous versions currently. We could imagine defining a `VersionedBitswap` transport that includes indicators of min/max supported version as it's payload.